### DISCUSSION TITLE
Problemas com rota

### DISCUSSION THREAD

**Discussion by fbonebrio** [balta-io/forum#528] [Tira Dúvidas] [answered]
Balta tudo bem?
Terminei o curso de Fundamentos do ASP.NET... e foi tudo certinho... inclusive paralelo fiz um projetinho a parte usando outro banco de dados e também deu certo....
Agora tentando usar ele novamente para estudo... começou a dar erro de restrição de rota....
InvalidOperationException: The constraint reference 'string' could not be resolved to a type. Register the constraint type with 'Microsoft.AspNetCore.Routing.RouteOptions.ConstraintMap'.
Fala sobre eu estar um padrão de rota que não está registrada na aplicação....
consegue me dar um norte para busca desse problema?

Muito Obrigado

**Comment by fbonebrio** [comment, marked as answer]
pessoal... consegui resolver....
estava assim
[HttpGet("/v1/companies/{cnpj:string}")]
agora está assim e funciona
[HttpGet("/v1/companies/{cnpj}")]

**Reply by andrebaltieri** [reply]
Boaaaa @fbonebrio!

Muito obrigado por compartilhar a solução 🚀
